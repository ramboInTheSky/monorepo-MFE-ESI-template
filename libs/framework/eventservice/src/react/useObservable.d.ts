import { ESB } from "../esb";
export declare const useCommonObservable: (esb: ESB, callback: (data: any) => void) => void;
