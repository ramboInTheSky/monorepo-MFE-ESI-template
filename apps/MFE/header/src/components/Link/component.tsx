import styled from "styled-components"

export const Anchor = styled.a`
    align-items: center;
    color: inherit;
    font-size: 12px;
    text-decoration: none;
    text-transform: capitalize;
    display: flex;
    align-items: center;
    letter-spacing: 0.0075rem;
    &:hover {
        text-decoration: underline;
    }
`
