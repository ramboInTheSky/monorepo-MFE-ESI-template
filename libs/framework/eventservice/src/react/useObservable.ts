import {useEffect} from "react"
import {ESB} from "../esb"

export const useCommonObservable = (esb: ESB, callback: (data: any) => void) => {
    return useEffect(() => {
        const {subscription} = esb.subscribe(callback)
        return () => {
            subscription.unsubscribe()
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
}
