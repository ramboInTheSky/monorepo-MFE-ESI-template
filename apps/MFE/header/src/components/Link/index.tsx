import React, {ReactNode} from "react"
import {Anchor} from "./component"

export type LinkProps = {
    href: string
    children: ReactNode
    dataGaV1?: string
    dataGaV2?: string
    rel?: string
    dataTestId?: string
}

const Link = ({href, children, dataGaV1, dataGaV2, rel, dataTestId}: LinkProps) => {
    const linkProps = {
        href,
        "data-ga-v1": dataGaV1,
        "data-ga-v2": dataGaV2,
    }

    // eslint-disable-next-line dot-notation
    if (rel) linkProps["rel"] = rel

    return (
        <Anchor {...linkProps} data-testid={dataTestId}>
            {children}
        </Anchor>
    )
}

export default Link
