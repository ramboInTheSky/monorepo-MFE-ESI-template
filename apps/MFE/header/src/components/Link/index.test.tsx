import React from "react"
import {render, screen} from "@testing-library/react"
import Link, {LinkProps} from "."

describe("Components - Link: ", () => {
    let props: Omit<LinkProps, "children">
    beforeEach(() => {
        props = {
            href: "https://sampleref",
        }
    })
    it("should match snapshot", () => {
        const {asFragment} = render(<Link {...props}> fake</Link>)
        expect(asFragment).toMatchSnapshot()
    })
    it("should have required attributes", () => {
        const content = "fake"
        render(<Link {...props}> {content}</Link>)
        const got = screen.getByRole("link")
        expect(got).toHaveAttribute("href", props.href)
        expect(got).toHaveTextContent(content)
    })

    it("should have optional rel attribute", () => {
        const content = "fake 2"
        render(
            <Link {...props} rel="nofollow">
                {" "}
                {content}
            </Link>,
        )
        const got = screen.getByRole("link")
        expect(got).toHaveAttribute("rel", props.rel)
        expect(got).toHaveTextContent(content)
    })
})
