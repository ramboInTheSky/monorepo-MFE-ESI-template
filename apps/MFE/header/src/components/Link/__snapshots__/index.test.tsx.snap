// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Link:  should match snapshot 1`] = `[Function]`;
